azure-quantum/azure/quantum/optimization/term.py

```diff
@@ -9,20 +9,52 @@
 
 __all__ = ['Term']
 
+try:
+    import numpy.typing as npt
+    WArray = Union[int, float, npt.ArrayLike]
+
+    def _convert_if_numpy_type(param : WArray):
+        # Attempt first a conversion to a supported type if parameter is a numpy float/int.
+        numpy_integer_types = [np.byte, np.ubyte, np.short, np.ushort, np.intc, np.uintc,
+                               np.int_, np.uint, np.longlong, np.ulonglong, np.int8, np.uint8,
+                               np.int16, np.uint16, np.int32, np.uint32, np.int64, np.uint64]
+
+        numpy_float_types = [np.float16, np.float32, np.float64, np.float_, np.half, np.single, np.double]
+
+        if (type(param) in numpy_integer_types + numpy_float_types):
```

Comment:
I'd also suggest dropping the parens around the conditional here, as per PEP 8 and/or black style guides.